aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy

```diff
@@ -66,29 +66,31 @@ abstract class AbstractSourceGeneratorSpec extends Specification {
     void generate() {
         def sourceGenerator = newGenerator()
         sourceGenerator.generate(context)
-        def sources = context.getGeneratedJavaFiles().collectEntries([:]) {
+        def javaFiles = context.getGeneratedJavaFiles()
+        def sources = new HashMap<JavaFile, String>()
+        javaFiles.forEach {
             def writer = new StringWriter()
             it.writeTo(writer)
-            [it, writer.toString()]
+            sources.put(it, writer.toString())
         }
 
         this.generatedSources = new GeneratedSources(sources, context.getGeneratedStaticInitializers(), resourcesDir.toFile())
     }
 
     void assertThatGeneratedSources(@DelegatesTo(value=GeneratedSources, strategy = Closure.DELEGATE_FIRST) Closure<?> spec) {
         if (generatedSources == null) {
-            throw new AssertionError("Couldn't find generated sources. Did you call 'generateSources()'?")
+            throw new AssertionError((Object) "Couldn't find generated sources. Did you call 'generateSources()'?")
```

Comment:
suggestion: I would prefer you revert those changes. There is no problem with the original code, I can only think you're doing this because the IDE messes up, but it shouldn't.

Reply:
Hmm,, maybe it was a bug in IDEA, but it's highlighted these lines in red for me, because it was not possible to determine the correct constructor and she indicated that the constructor with the string is private. Okay, I'll remove these changes. 
![изображение](https://user-images.githubusercontent.com/48474279/221603500-0a024e61-0fae-4c19-9d83-de94083b47bb.png)

Reply:
Yes it's a false positive in IDEA